service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskErrorHandler.java

```diff
@@ -24,45 +50,53 @@ public boolean handleRequestError(TaskDefinition taskDefinition) {
         );
     }
 
-    public boolean handleTranslationError(TaskDefinition taskDefinition) {
+    private boolean handleTranslationError(TaskDefinition taskDefinition) {
         return handleFailingProcess(
             taskDefinition,
             "10",
             "Failed to successfully generate EHR Extract."
         );
     }
 
-    public boolean handleGeneralProcessingError(TaskDefinition taskDefinition) {
+    private boolean handleGeneralProcessingError(TaskDefinition taskDefinition) {
         return handleFailingProcess(
             taskDefinition,
             "99",
             "An error occurred when executing a task"
         );
     }
 
-    public boolean handleGpConnectError(TaskDefinition taskDefinition) {
+    private boolean handleGpConnectError(TaskDefinition taskDefinition) {
         return handleFailingProcess(
             taskDefinition,
             "20",
             "Spine system responded with an error"
         );
     }
 
-    public boolean handleInvalidNotAuthError(TaskDefinition taskDefinition) {
+    private boolean handleInvalidNotAuthError(TaskDefinition taskDefinition) {
         return handleFailingProcess(
                 taskDefinition,
                 "19",
                 "Sender check indicates that Requester is not the patient’s current healthcare provider"
         );
     }
-    public boolean handleNotFoundError(TaskDefinition taskDefinition) {
+    private boolean handleNotFoundError(TaskDefinition taskDefinition) {
         return handleFailingProcess(
                 taskDefinition,
                 "06",
                 "Patient not at surgery."
         );
     }
 
+    private boolean handleMaximumExternalAttachmentsError(TaskDefinition taskDefinition) {
+        return handleFailingProcess(
+            taskDefinition,
+            "30",
+            "Large Message general failure"
```

Comment:
I'm not sure how this message is used, but it currently feels quite vague.

If we know this is specifically about "too many attachments for the MHS adaptor", including that in the string might be useful, depending on how this string is used.

Reply:
Agreed, I took the text from [this document](file:///C:/Users/BenjaminHession/Downloads/NPFIT-PC-BLD-0083.08%20GP2GP%20Response%20Codes%20(1).pdf), but we are using it in a different context here. I'll update it. The values are inserted into the Negative Acknowledgement, however the string is optional. So I think we should be ok to change it. The code is the more important element for the receiving system.